File: WebApplication13/Models/Mobile/MobileUserInfo.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace WebApplication13.Models.Mobile {
    public class MobileUserInfo
{
        public string Token { get; set; }
        public string Username { get; set; }
        public string Password { get; set; }
        public string TitleName { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string IdCardNumber { get; set; }
        public string Nationality { get; set; }
        public Nullable<System.DateTime> Birthday { get; set; }
        public string Address { get; set; }
        public Nullable<int> Province { get; set; }
        public Nullable<int> Occupation { get; set; }
        public string PhoneNumber { get; set; }
        public string Email { get; set; }
        public string LineId { get; set; }
        public string WhatsAppId { get; set; }
        public Nullable<int> CompanyTypeOfUsage { get; set; }
        public string CompanyName { get; set; }
        public string CompanyTaxId { get; set; }
        public string CompanyAddress { get; set; }
        public Nullable<int> Bank { get; set; }
        public string BankAccountNumber { get; set; }
        public string ProfilePath { get; set; }
        public string Active { get; set; }
        public string TierName { get; set; }
        public string TierColor { get; set; }
        public double TotalBaht { get; set; }
        public double MaxBaht { get; set; }
    }
}

File: spa_commission/lib/models/mobile_user_info.dart
import 'dart:convert';

class MobileUserInfo {
  String? Token;
  String? Username;
  String? FirstName;
  String? LastName;
  String? IdCardNumber;
  String? Nationality;
  DateTime? Birthday;
  String? Address;
  String? Province;
  String? Occupation;
  String? PhoneNumber;
  String? Email;
  String? LineId;
  String? WhatsAppId;
  String? CompanyTypeOfUsage;
  String? CompanyName;
  String? CompanyTaxId;
  String? CompanyAddress;
  String? BankAccount;
  String? BankAccountNumber;
  String? ProfilePath;
  String? TierName;
  String? TierColor;
  double TotalBaht;
  double MaxBaht;

  MobileUserInfo({
    this.Token,
    this.Username,
    this.FirstName,
    this.LastName,
    this.IdCardNumber,
    this.Nationality,
    this.Birthday,
    this.Address,
    this.Province,
    this.Occupation,
    this.PhoneNumber,
    this.Email,
    this.LineId,
    this.WhatsAppId,
    this.CompanyTypeOfUsage,
    this.CompanyName,
    this.CompanyTaxId,
    this.CompanyAddress,
    this.BankAccount,
    this.BankAccountNumber,
    this.ProfilePath,
    this.TierName,
    this.TierColor,
    this.TotalBaht = 0,
    this.MaxBaht = 0,
  });

  factory MobileUserInfo.fromJson(String json) {
    Map<String, dynamic> jsonMap = jsonDecode(json);
    String? _birthday = jsonMap['Birthday'];
    DateTime? birthday;
    if (_birthday != null) birthday = DateTime.parse(_birthday);
    return MobileUserInfo(
      Token: jsonMap['Token'],
      Username: jsonMap['Username'],
      FirstName: jsonMap['FirstName'],
      LastName: jsonMap['LastName'],
      IdCardNumber: jsonMap['IdCardNumber'],
      Nationality: jsonMap['Nationality'],
      Birthday: birthday,
      Address: jsonMap['Address'],
      Province: jsonMap['Province'],
      Occupation: jsonMap['Occupation'],
      PhoneNumber: jsonMap['PhoneNumber'],
      Email: jsonMap['Email'],
      LineId: jsonMap['LineId'],
      WhatsAppId: jsonMap['WhatsAppId'],
      CompanyTypeOfUsage: jsonMap['CompanyTypeOfUsage'],
      CompanyName: jsonMap['CompanyName'],
      CompanyTaxId: jsonMap['CompanyTaxId'],
      CompanyAddress: jsonMap['CompanyAddress'],
      BankAccount: jsonMap['BankAccount'],
      BankAccountNumber: jsonMap['BankAccountNumber'],
      ProfilePath: jsonMap['ProfilePath'],
      TierName: jsonMap['TierName'],
      TierColor: jsonMap['TierColor'],
      TotalBaht: jsonMap['TotalBaht'],
      MaxBaht: jsonMap['MaxBaht'],
    );
  }
}
